core/embed/sec/optiga/optiga_commands.c

```diff
@@ -515,6 +533,14 @@ optiga_result optiga_set_auto_state(uint16_t nonce_oid, uint16_t key_oid,
   hmac_sha256(key, key_size, nonce, sizeof(nonce), ptr);
 
   OPTIGA_LOG(__func__, tx_buffer, tx_size)
+  // The auto-state is added before the command is actually executed. Otherwise,
+  // it could happen that the command succeeded, but an error occurred during
+  // the receipt of the response. In such a case, the auto-state couldn't be
+  // cleared using optiga_clear_all_auto_states().
+  if (!auto_states_add(key_oid)) {
+    // This should never happen.
+    return OPTIGA_ERR_UNEXPECTED;
```

Comment:
It can happen if too many auto states are requested. We should probably return what Optiga would return `OPTIGA_ERR_CODE_MEMORY`.